src/combinators/mod.rs

```diff
@@ -97,34 +106,13 @@ pub fn hex<S: ::lib::std::fmt::UpperHex, W: Write>(data: S) -> impl SerializeFn<
 ///
 /// let mut buf = [0u8; 100];
 ///
-/// let (out, _) = skip(2)(&mut buf[..]).unwrap();
+/// let out = skip(2)(&mut buf[..]).unwrap();
 ///
 /// assert_eq!(out.len(), 98);
 /// ```
-pub fn skip<W: Skip>(len: usize) -> impl SerializeFn<W> {
+pub fn skip<W: Write + Skip>(len: usize) -> impl SerializeFn<W> {
     move |out: W| {
-        out.skip(len).map(|out| (out, len))
-    }
-}
-
-/// applies a serializer then returns a tuple containing what was written and the remaining output buffer
-///
-/// ```rust
-/// use cookie_factory::{position, string};
-///
-/// let mut buf = [0u8; 100];
-///
-/// let (written, remaining) = position(string("abcd"))(&mut buf[..]).unwrap();
-///
-/// assert_eq!(remaining.len(), 96);
-/// assert_eq!(written, &b"abcd"[..]);
-/// ```
-pub fn position<'a, F>(f: F) -> impl Fn(&'a mut [u8]) -> Result<(&'a mut [u8], &'a mut [u8]), GenError>
```

Comment:
Not sure if we want to drop that or not.
I left this untouched for now in my premious patches

Reply:
I can put it back, it just seems inconsistent with everything else in this module. It only works on byte slices, does not return `SerializeFn` but some strange other kind of function, it does not advance the input byte slice but returns things, ...

Reply:
I was tempted to do so too, let’s nuke it

Reply:
:+